src/client/src/orangehrmBuzzPlugin/components/BasePost.vue

```diff
@@ -1,33 +1,126 @@
+        <div class="orangehrm-buzz-post-header-details">
+          <div class="orangehrm-buzz-post-profile-image">
+            <img
+              alt="profile picture"
+              class="employee-image"
+              :src="`../pim/viewPhoto/empNumber/${employeeId}`"
+            />
+          </div>
+          <div class="orangehrm-buzz-post-header-text">
+            <oxd-text tag="p" class="orangehrm-buzz-post-emp-name">
+              {{ employeeFullName }}
+            </oxd-text>
+            <oxd-text tag="p" class="orangehrm-buzz-post-time">
+              {{ postedTime }}
+            </oxd-text>
+          </div>
+        </div>
+        <div
+          v-if="postedEmployeeId === buzzUserId"
+          class="orangehrm-buzz-post-header-config"
+        >
+          <oxd-icon-button
+            class=""
+            name="three-dots"
+            with-container="true"
+            @click="onClickShowDropdown"
+            @focusout="showDropdown = false"
+          />
+          <oxd-sheet
+            v-if="showDropdown"
+            class="orangehrm-buzz-post-header-dropdown"
+          >
+            <div class="orangehrm-buzz-post-header-dropdown-item">
+              <oxd-icon name="trash" />
+              <oxd-text tag="p">
+                {{ $t('buzz.delete_post') }}
+              </oxd-text>
+            </div>
+            <oxd-divider />
+            <div class="orangehrm-buzz-post-header-dropdown-item">
+              <oxd-icon name="pencil" />
+              <oxd-text tag="p">
+                {{ $t('buzz.edit_post') }}
+              </oxd-text>
+            </div>
+          </oxd-sheet>
+        </div>
+      </div>
+      <oxd-divider />
+    </div>
+    <div class="orangehrm-buzz-post-body">
+      <oxd-text>{{ content }}</oxd-text>
+      <slot name="body"></slot>
+    </div>
+    <div class="orangehrm-buzz-post-footer">
+      <div class="orangehrm-buzz-post-footer-left">
+        <slot name="actionButton"> </slot>
+      </div>
+      <slot name="postStatus"> </slot>
     </div>
-    <oxd-divider></oxd-divider>
-    <oxd-text type="card-title">{{ content }}</oxd-text>
+    <div v-if="showComments" class="orangehrm-buzz-post-comment">
+      <oxd-divider />
+      <div class="orangehrm-buzz-post-comment-add">
+        <div class="orangehrm-buzz-post-profile-image-comment">
+          <img
+            alt="profile picture"
+            class="employee-image"
+            :src="`../pim/viewPhoto/empNumber/${buzzUserId}`"
+          />
+        </div>
+        <oxd-form class="orangehrm-buzz-post-comment-input">
+          <oxd-form-row>
+            <oxd-grid-item>
+              <oxd-input-field :placeholder="$t('buzz.write_your_comment')" />
+            </oxd-grid-item>
+          </oxd-form-row>
+        </oxd-form>
+      </div>
+      <!-- bottom slot for commnets -->
+      <slot name="comments"></slot>
+    </div>
+    <delete-confirmation ref="deleteDialog"></delete-confirmation>
```

Comment:
delete-confirmation is irrelevant to base-post (and it will get repeated in entire news feed as base post is iterated). base post should only facilitate rendering of the post, all actions should be emitted to parent and done at parent level.